ssl/ssl_transfer_asn1.cc

```diff
@@ -435,14 +853,65 @@ static int SSL3_STATE_from_bytes(SSL *ssl, CBS *cbs, const SSL_CTX *ctx) {
   OPENSSL_memcpy(out->client_random, CBS_data(&client_random),
                  SSL3_RANDOM_SIZE);
   SSL_SESSION *session = out->established_session.get();
-  // the impl of |SSL_serialize_handback|, which only fetch IV when it's TLS 1.
-  Array<uint8_t> key_block1, key_block2;
-  if (!tls1_configure_aead(ssl, evp_aead_seal, &key_block1, session, {})) {
+  if (is_tls13) {
+    if (!tls13_set_traffic_key(ssl, ssl_encryption_application, evp_aead_seal,
+                               session,
+                               MakeSpan(CBS_data(&write_traffic_secret),
+                                        write_traffic_secret_len))) {
+      return 0;
+    }
+    if (!tls13_set_traffic_key(ssl, ssl_encryption_application, evp_aead_open,
+                               session,
+                               MakeSpan(CBS_data(&read_traffic_secret),
+                                        read_traffic_secret_len))) {
+      return 0;
+    }
+
+    // !! We set pending_hs_data and pending_hs_flight after as to avoid
```

Comment:
maybe we need another function: `tls13_set_traffic_key_no_flush()`. Just thoughts.

Reply:
It's not a bad idea, I had thought about actually exposing some of the flushing capabilities in general to avoid having to serialize these two pieces of data. The trade-off that came to mind in those scenarios was the added complexity to the upstream merge if we shift function definitions around to provide implementations that allow this sort of flow control and/or needing to duplicate logic to expose said functionality.